Bottom: 5b5e9db3d94df4c32490b5dc29c887400922bf3f
Top:    762d72427b7f4ea52aa7ee2952ab58b78dc26b8b
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-07 00:31:02 +0100

GTK HID TEST


---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 03fa2da..7629b6f 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -446,12 +446,10 @@ ghid_port_button_press_cb (GtkWidget * drawing_area,
   state = (GdkModifierType) (ev->state);
   mk = ghid_modifier_keys_state (&state);
   ghid_show_crosshair (FALSE);
-  notify_crosshair_change (false);
 
   do_mouse_action(ev->button, mk);
 
   ghid_invalidate_all ();
-  notify_crosshair_change (true);
   ghid_window_set_name_label (PCB->Name);
   ghid_set_status_line_label ();
   ghid_show_crosshair (TRUE);
@@ -472,13 +470,12 @@ ghid_port_button_release_cb (GtkWidget * drawing_area,
   state = (GdkModifierType) (ev->state);
   mk = ghid_modifier_keys_state (&state);
 
-  notify_crosshair_change (false);
-
   do_mouse_action(ev->button, mk + M_Release);
 
+  notify_crosshair_change (false);
   AdjustAttachedObjects ();
-  ghid_invalidate_all ();
   notify_crosshair_change (true);
+  ghid_invalidate_all ();
 
   ghid_window_set_name_label (PCB->Name);
   ghid_set_status_line_label ();
diff --git a/src/hid/gtk/gui-top-window.c b/src/hid/gtk/gui-top-window.c
index b2f0643..e21ef18 100644
--- a/src/hid/gtk/gui-top-window.c
+++ b/src/hid/gtk/gui-top-window.c
@@ -652,9 +652,8 @@ ghid_menu_cb (GtkAction * action, gpointer data)
     {
       notify_crosshair_change (false);
       AdjustAttachedObjects ();
-      ghid_invalidate_all ();
       notify_crosshair_change (true);
-      ghid_screen_update ();
+      ghid_invalidate_all ();
       ghid_window_set_name_label (PCB->Name);
       ghid_set_status_line_label ();
 #ifdef FIXME
